### ISSUE TITLE
Different values in Target and Prediction tensor

### ISSUE THREAD

**Issue by jessika0991** [facebook/fb.resnet.torch#77]
Hello,

I was trying to fine-tune one of the pretrained model. I have reset the classifier to a binary one. I printed out the prediction and the target tensors and I realized they actually contain different values. What I mean whit this is that in the Prediction tensor I find either 1 or 2, while in the Target tensor values are either 2 or 3. This clearly ends up in misclassification.

How can I solve the problem? 

Thanks in advance,
Jessica

**Comment by colesbury** [comment]
I don't understand: how are the target tensor values 2 or 3 when you're doing binary classification?

Can you describe what you're doing a little bit more?

**Comment by jessika0991** [comment]
Thanks for your reply.

This is the command I give:
th main.lua -retrain resnet-50.t7 -data path/to/my/data -resetClassifier true -nClasses 2 

I stored my data in a folder containing val and train subfolders. In turn, they are containing two other subfolders.
What exactly do you need to know?

**Comment by colesbury** [comment]
Perhaps there was an extra folder when the train and val directories were processed. Try deleting the `gen` directory.

After deleting `gen`, you may want to add extra debugging info to `imagenet-gen.lua`

Add: `print('classToIdx', classToIdx)` here https://github.com/facebook/fb.resnet.torch/blob/master/datasets/imagenet-gen.lua#L100

When you re-run your training command, you should see the mapping of directory name to label, which should only be one and two.

**Comment by jessika0991** [comment]
I solved the problem this way. There was an hidden folder that was taken into consideration as well
